torch/utils/data/datapipes/iter/ziparchivereader.py

```diff
@@ -25,42 +24,36 @@ class ZipArchiveReaderIterDataPipe(IterDataPipe[Tuple[str, BufferedIOBase]]):
     """
     def __init__(
             self,
-            datapipe: Iterable[Tuple[str, BufferedIOBase]],
+            datapipe: Iterable[str],
             length: int = -1):
         super().__init__()
-        self.datapipe: Iterable[Tuple[str, BufferedIOBase]] = datapipe
+        self.datapipe: Iterable[str] = datapipe
         self.length: int = length
 
     def __iter__(self) -> Iterator[Tuple[str, BufferedIOBase]]:
         if not isinstance(self.datapipe, Iterable):
-            raise TypeError("datapipe must be Iterable type but got {}".format(type(self.datapipe)))
-        for data in self.datapipe:
-            validate_pathname_binary_tuple(data)
```

Comment:
Is there any place we are still using `validate_pathname_binary_tuple`?

Reply:
Aside from the second PR, no. I can remove it in the second PR.

Reply:
After this stack of PR is landed. Could you please also modify the corresponding DataPipe (TarArchiveReader, ZipArchiveReader, XzFileReader) and text examples in `torchdata`

https://github.com/facebookexternal/torchdata/blob/537c4ae0d76e1cb983d7ff84037e792e2d4fba45/datapipes/iter/util/xzfilereader.py#L38

Reply:
Will do!